test/functorch/test_aotdispatch.py

```diff
@@ -2301,10 +2301,19 @@ def test_aot_autograd_symbolic_exhaustive(self, device, dtype, op):
                                 # of a tracing tensor with aten._local_scalar_dense.default -
                                 # erroring out! It's likely that this is caused by data-dependent
                                 # control flow or similar.
+    torch.nn.TransformerEncoder,  # DataDependentOutputException: aten.equal compares a mask input
+                                  # to a causal mask tensor, to see if Boolean is_causal should be set
+                                  # for TrnasformerEncoder layers, MHA and sdp custom kernels
+    torch.nn.Transformer,  # DataDependentOutputException: aten.equal compares a mask input
```

Comment:
Yes, it is - because we're now recognizing a causal mask and converting it to the is_causal boolean.  Since it depends on the mask input, it's input dependent evaluation.